k-distribution/include/kframework/cparser/main.c

```diff
@@ -1,37 +1,102 @@
+#include "node.h"
+#include "parser.tab.h"
+#include "scanner.h"
+
+#include <assert.h>
+#include <stdarg.h>
 #include <stdbool.h>
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
-#include "node.h"
-#include "parser.tab.h"
-#include "scanner.h"
+
+struct string_buffer {
+  FILE *fp;
+  char *buf;
+  size_t capacity;
+  size_t idx;
+};
```

Comment:
I'm not very familiar with C, but this seems like there should be something implemented already.

Reply:
Yeah, the problem is managing dependencies in C - there's similar code in the LLVM backend, but reusing that or using a library would require far more code than just implementing the type here.